pystoned/CNLS.py

```diff
@@ -1,304 +1,224 @@
-
-                # production model
-                if fun == "prod":
-
-                    def qconcav_rule(model, i, h):
-                        arow = x[i]
-                        if i == h:
-                            return Constraint.Skip
-                        return model.a[i] + sum(model.b[i, j] * arow[j] for j in model.j) <= model.a[h] + \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconcav = Constraint(model.i, model.h, rule=qconcav_rule, doc='concavity constraint')
-
-                # cost model
-                if fun == "cost":
-
-                    def qconvex_rule(model, i, h):
-                        arow = x[i]
-                        if i == h:
-                            return Constraint.Skip
-                        return model.a[i] + sum(model.b[i, j] * arow[j] for j in model.j) >= model.a[h] + \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconvex = Constraint(model.i, model.h, rule=qconvex_rule, doc='convexity constraint')
-
-            if rts == "crs":
-
-                # Constraints
-                def qreg_rule(model, i):
-                    return log(y[i]) == log(model.f[i] + 1) + model.e[i]
-
-                model.qreg = Constraint(model.i, rule=qreg_rule, doc='log-transformed regression equation')
-
-                def qlog_rule(model, i):
-                    arow = x[i]
-                    return model.f[i] == sum(model.b[i, j] * arow[j] for j in model.j) - 1
-
-                model.qlog = Constraint(model.i, rule=qlog_rule, doc='regression function')
-
-                # production model
-                if fun == "prod":
-
-                    def qconcav_rule(model, i, h):
-                        arow = x[i]
-                        if i == h:
-                            return Constraint.Skip
-                        return sum(model.b[i, j] * arow[j] for j in model.j) <= \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconcav = Constraint(model.i, model.h, rule=qconcav_rule, doc='concavity constraint')
-
-                # cost model
-                if fun == "cost":
-
-                    def qconvex_rule(model, i, h):
-                        arow = x[i]
-                        if i == h:
-                            return Constraint.Skip
-                        return sum(model.b[i, j] * arow[j] for j in model.j) >= \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconvex = Constraint(model.i, model.h, rule=qconvex_rule, doc='convexity constraint')
-
-    return model
\ No newline at end of file
+# Import pyomp module
+from pyomo.environ import *
+from pyomo.core.expr.numvalue import NumericValue
+
+
+class CNLS:
+    def __init__(self, y, x, cet='addi', fun='prod', rts='vrs'):
```

Comment:
And a new comment here.
`"""Convex Nonparametric Least Square (CNLS)"""`

Reply:
Yes, this should be added.

Reply:
OK.